src/DeclarativeWebPushMessage.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+namespace NotificationChannels\WebPush;
+
+use Illuminate\Support\Arr;
+use NotificationChannels\WebPush\Exceptions\MessageValidationFailed;
+
+/**
+ * @implements \Illuminate\Contracts\Support\Arrayable<string, mixed>
+ *
+ * @link https://www.w3.org/TR/push-api/#members
+ */
+class DeclarativeWebPushMessage implements WebPushMessageInterface
+{
+    protected string $title;
+
+    /**
+     * @var array<array-key, array{'title': string, 'action': string, 'navigate': string, 'icon'?: string}>
+     */
+    protected array $actions = [];
+
+    protected string $badge;
+
+    protected string $body;
+
+    protected string $dir;
+
+    protected string $icon;
+
+    protected string $image;
+
+    protected string $lang;
+
+    protected bool $mutable;
+
+    protected string $navigate;
+
+    protected bool $renotify;
+
+    protected bool $requireInteraction;
+
+    protected bool $silent;
+
+    protected string $tag;
+
+    protected int $timestamp;
+
+    /**
+     * @var array<int>
+     */
+    protected array $vibrate;
+
+    protected mixed $data;
+
+    /**
+     * @var array<string, mixed>
+     */
+    protected array $options = [
+        'contentType' => 'application/json',
```

Comment:
[In the documentation](https://github.com/WebKit/explainers/blob/main/DeclarativeWebPush/README.md#rfc-8030---generic-event-delivery-using-http-push) I saw `application/notification+json`, but not sure if this is now a requirement?